### PR TITLE
Add Federation changelog to our docs

### PR DESCRIPTION
**Pull request by patrick91**
This PR is based on @StephenBarlow initial work on adding a federation changelog. At the moment the aim is make it easier for subgraph implementers to see what has changed, but we can extend the scope 😊

This is how the page looks like at the moment:

CleanShot 2023-02-08 at 17 45 31@2x

Closes #2076

### CODE REVIEW

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:16]: technically definition already defined interfaces as a valid location (but it didn't work) -> v2.3 actually fixes that

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:23]: v2 also makes `@shareable` repeatable

**Comment by dariuszkuc** [comment]
question -> who is the target audience? subgraph developers? subgraph library maintainers? other?

File: docs/source/federation-versions.mdx
Comment (patrick91) [reply docs/source/federation-versions.mdx:16]: do you think it is worth mentioning anyway, or shall I drop it?

**Comment by patrick91** [comment]
> question -> who is the target audience? subgraph developers? subgraph library maintainers? other?

I was thinking library maintainers first, then we can expand :)

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [reply docs/source/federation-versions.mdx:16]: there is no change required on the lib side so don't think there is a need for it. I'd call out that when extending interfaces through`@interfaceObject` types, that subgraph have to handle the "interface" __typename in the _entities resolver

File: docs/source/federation-versions.mdx
Comment (StephenBarlow) [inline docs/source/federation-versions.mdx:103]:
```suggestion
  resolvable: Boolean = true # highlight-line
```

**Review by StephenBarlow — APPROVED** [review]
Some small suggestions here. I think once this is merged, I'd like to take a stab at formatting the _newer_ entries to match the tabular format used by the older ones for consistency's sake, but I see no issue with this version going out first.

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:17]: this directive already specified interface location in earlier version but it never worked (I think it was originally ignored but later on started to fail?)

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:29]:
```suggestion
The `@shareable` directive has been [updated to be repeatable](https://github.com/apollographql/federation/pull/2175). This allows you to use in a type definition and in an extension of that type definition.
```

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:10]: should those be bulleted items? i.e. it is unclear whether `@key` on interface definitions is separate item or not?

Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:10]:
```suggestion
* introduce [`@interfaceObject`](https://www.apollographql.com/docs/federation/federated-types/interfaces) directive - a directive that allows you to abstract an interface in isolation by declaring it locally as an object type. This enables you to contribute fields without needing to redefine all the interface’s concrete implementation types.
```

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:16]:
```suggestion
* support `@key` on interface definitions - while `@key` directive definition already defined `INTERFACE` as a valid location, this information wasn't used during composition. With introduction of `@interfaceObject` we now need to specify matching `@key` on abstract interfaces
```

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:49]: you provide an example here on how to use compose directive? should we have similar example for `@interfaceObject`?

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:43]:
```suggestion
* introduce new [`@composeDirective`](https://www.apollographql.com/docs/federation/federated-types/federated-directives/#composedirective) - 
directive indicates to composition that all uses
of a particular custom type system directive in the subgraph schema should be preserved
in the supergraph schema (by default, composition omits most directives from the supergraph schema).
```

File: docs/source/federation-versions.mdx
Comment (dariuszkuc) [inline docs/source/federation-versions.mdx:67]: this is also true for `@interfaceObject`

Comment (patrick91) [inline docs/source/federation-versions.mdx:67]:
```suggestion
**Note:** like for federation directives, make sure to include the `@composeDirective` directive in the `import` list of the `@link` directive that points to the spec that defines the directive. You can see in our example above that, in addition to importing the federation directives, we are also importing `@myDirective` from `"https://myspecs.dev/myDirective/v1.0"`
```

File: docs/source/federation-versions.mdx
Comment (StephenBarlow) [inline docs/source/federation-versions.mdx:28]:
````suggestion
<td>

Introduced. [Learn more.](/federation/federated-types/interfaces)

```graphql
````

File: docs/source/federation-versions.mdx
Comment (StephenBarlow) [inline docs/source/federation-versions.mdx:58]: I would instead make this a second _directive_ change in the table _above_ for simplicity, where the left column value is `@key` and the description is "Composition now allows `@key` to be applied to interface definitions."

File: docs/source/federation-versions.mdx
Comment (StephenBarlow) [inline docs/source/federation-versions.mdx:121]: I might also fold this info into the table entry above for `@shareable` so it says:

> Added `repeatable` to the directive definition. Additionally, composition now rejects `@shareable` when applied to interface fields.

File: docs/source/federation-versions.mdx
Comment (StephenBarlow) [inline docs/source/federation-versions.mdx:153]:
````suggestion
<td>

Introduced. [Learn more.](/federation/federated-types/federated-directives#composedirective)

```graphql
````

**Review by StephenBarlow — APPROVED** [review]
Looking great! Approved with some suggestions

**Pull request merged**
